frontend create account pages + created root style

throwExceptionPlease opened the pull request:
added frontend aspects for the create account pages. also added CSS variables and root styles. all create-account pages are in src > create-acc-pages and the logo was added in an assets folder.

ardada2468 commented:
Hey,

The UI looks great but I think we may be changing the login page based on the meeting yesterday. But while the designer are working on the changes could you fix the following: 

- I see that you are checking if the password meets the requirements but I cannot see the checkbox UI password validation on the latest figma design. 
- The form does not do email validation, I was able to put in any random string and it was accepting it.

throwExceptionPlease commented:
Added checking for both password fields to contain the same value; added check to prevent user from submitting form without satisfying the pwd reqs; changed input type from text to email for email inputs (oops); updated colors in root style as per figma; added checkbox ui password validation in both pages

not yet implemented: adding the icon and red border for invalid confirmation password, and new styling for when a pwd req has yet to be fulfilled

will work on some more later tonight

throwExceptionPlease commented:
Finished with everything except one, which is getting the minus sign in the checkbox

aduvvuri007 commented:
I added all the backend code onto this branch to make it easier. I added the function to create accounts with email and password, added user data to the firestore database and their specified collection(applicants and reviewers) and then I created the cloud functions to add custom claims to users (functions not deployed yet and still not tested)

ardada2468 commented on react-app/ccf/src/create-acc-pages/create-acc-reviewer/CreateAccReviewer.tsx:
```diff
@@ -0,0 +1,261 @@
+      .catch((error) => {
+        console.log('Error: ', error);
+      });
+      console.log(user);
```
We should not be printing this information in the console.

ardada2468 reviewed (changes requested):
Great Work Guys, This is almost ready to be merged but just needs a couple of edits: 

1. We need to do email validation because I can still hit the sumbit button when i only put ardada2468 (error shows up in console).

![image](https://github.com/user-attachments/assets/72896ff5-dc14-423a-8275-d772fb1eecff)

2. Remove all instances where we are console.logging the user information from firebase. 

![image](https://github.com/user-attachments/assets/cd391b7a-0396-4f70-aa15-7c1a12f8feef)

jdlane commented on functions/src/index.ts:
```diff
@@ -17,3 +20,32 @@ import * as logger from "firebase-functions/logger";
+    }
+    ).then(() => {
+        return {
+            message: `Success! ${data.email} has been made an admin.`
```
Small thing, this returns "has been made an admin" for any user role. It should say "reviewer" for reviewer etc.

throwExceptionPlease commented:
Added email validation. There was no design (and technical) guide for indicating an invalid email, so I just reused what was done for the password.

ardada2468 reviewed (changes requested):
This fixes all of the issues but if you could just add 2 more things: 

After account creation, automatically redirect the user to the sign-in page.

Disable the "Submit" button (gray it out) if any required field is not completed or does not pass validation. The button should remain unclickable until:

- All fields are filled out.
- All entries have passed validation checks.

Once all fields are validated, the button becomes clickable, enabling the user to submit the form.

jdlane reviewed (commented):
We talked in the meeting about greying out the submit button until the fields are all valid. Could you add that change before we merge?

throwExceptionPlease commented:
added extra conditionals to the submit button to ensure no field is left behind or invalid (this ones for pwds + email) before the user is able to click on the submit button. not quite sure if u needed me to validate the first name, last name, and affiliation too so i left them be. didn't see any new updates on the login stuff when i did the pull, so i rerouted the submit button from the dashboards to the login page as specified in app.tsx

let me know if i missed anything.

jdlane reviewed (changes requested):
Nice job, just one last fix. We need the creating account logic in the CreateAcc files to be moved to separate functions in a new AccountManager file in the functions folder. That way we avoid repeated code and can easily find account functions to make any changes in the future. Also, instead of creating new functions and firestore objects, you should import the existing ones from index.ts

jdlane reviewed (commented):
Also, to keep things consistent we should probably either make a "create-acc-applicant" folder or move the create reviewer files out of their own folder.

throwExceptionPlease commented:
CreateAccApplicants and CreateAccReviewers are now in their separate folders in the create-acc-pages folder. I realized that createAccApplicants didn't have the pwd validation text that createAccReviewers had so I added it.

ardada2468 commented:
This is great, please fix the merge conflict and we are ready to merge.

jdlane commented:
> This is great, please fix the merge conflict and we are ready to merge.

I'm going to go ahead and resolve those myself, just so we get that through. Nice job guys!
